great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -709,6 +709,11 @@ def _add_fluent_datasource(self, datasource: FluentDatasource) -> None:
                 f"Can not write the fluent datasource {datasource.name} because a datasource of that "
                 "name already exists in the data context."
             )
+        if self._datasource_store.cloud_mode:
+            set_datasource = self._datasource_store.set(key=None, value=datasource)
+            if set_datasource.id:
+                logger.debug(f"Assigning `id` to '{datasource.name}'")
+                datasource.id = set_datasource.id
```

Comment:
Temporary workaround while we update `stores` to work better with Fluent Datasources.
Without this we end up with duplicate entries for fluent datasources in both `"fluent_datasources"` and `"datasources"` config entries.

Reply:
Could we add this context as a comment above the block?

Reply:
Will do.

Reply:
This seems to be syncing `datasource` with the returned `set_datasource`. It just happens that the only change is the id. Maybe this sync/copy method should live on datasource? This is fine for now, especially if it's changing soon per your comment.